### PR TITLE
SNOW-1706990 Convert `snow app validate` to v2-native

### PR DESCRIPTION
**Pull request by sfc-gh-fcampbell**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Converts `snow app validate` to operate on v2 entities natively.

### CODE REVIEW

File: tests/__snapshots__/test_help_messages.ambr [hunk 1]
```diff
@@ -917,18 +917,45 @@
...
-  | --app-entity-id              TEXT  The ID of the application entity on which |
-  |                                    to operate when definition_version is 2   |
-  |                                    or higher.                                |
-  | --project            -p      TEXT  Path where Snowflake project resides.     |
-  |                                    Defaults to current working directory.    |
-  | --env                        TEXT  String in format of key=value. Overrides  |
-  |                                    variables from env section used for       |
-  |                                    templates.                                |
-  | --help               -h            Show this message and exit.               |
+  | --interactive            --no-interactive          When enabled, this option |
```
Comment (sfc-gh-bdufour) [inline tests/__snapshots__/test_help_messages.ambr:920]: What does --interactive / --force mean for validate?

Comment (sfc-gh-bdufour) [reply tests/__snapshots__/test_help_messages.ambr:920]: For the scratch stage, we shouldn't really need interactive or force. So if that's the only reason to have them here, I'd be inclined to match the `app` command.

Comment (sfc-gh-fcampbell) [reply tests/__snapshots__/test_help_messages.ambr]: updated

File: src/snowflake/cli/_plugins/nativeapp/entities/application_package.py [hunk 2]
```diff
@@ -1243,8 +1243,38 @@ def validate_setup_script(
             if validation_result["status"] == "FAIL":
                 raise SetupScriptFailedValidation()
 
+    def get_validation_result(self, use_scratch_stage: bool = True):
+        model = self._entity_model
+        workspace_ctx = self._workspace_ctx
+        package_name = model.fqn.identifier
+        return self.get_validation_result_static(
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/_plugins/nativeapp/entities/application_package.py:1250]: this call pattern is concerning. Not a blocker, but let's discuss offline.

**Review by sfc-gh-bdufour — DISMISSED** [review]

**Review by sfc-gh-bdufour — APPROVED** [review]

**Pull request merged**